src/adagrams.js

```diff
@@ -1,15 +1,150 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  let playersHand = [];
+  const letterPool = getLetterPool(LETTER_POOL);
+
+  while (playersHand.length !== 10) {
+    let letter = randomLetter(letterPool);
+    playersHand.push(letter);
+  }
+
+  return playersHand;
 };
 
+// grab numbers and then make sure we can use that letter
+
 export const usesAvailableLetters = (input, lettersInHand) => {
+  input = input.toUpperCase();
+  let availableBank = counter(lettersInHand);
+  for (const letter of input) {
+    if (letter in lettersInHand || availableBank[letter]) {
+      availableBank[letter] -= 1;
+    } else {
+      return false;
+    }
+  }
+  return true;
   // Implement this method for wave 2
 };
 
 export const scoreWord = (word) => {
   // Implement this method for wave 3
-};
 
+  word = word.toUpperCase();
+
+  const ps = {};
+
+  ps.A = ps.E = ps.I = ps.O = ps.U = ps.L = ps.N = ps.R = ps.S = ps.T = 1;
```

Comment:
This is a valid means of assigning values to multiple properties in an object, however, it's not very easy to read and not a very common pattern in JS. Also, since the scores object doesn't need to be re-created every time `scoreWord` is called. So, it would be ideal to move it out into the main module scope.